### PR TITLE
Boost: Fix linting config and standards in files

### PR DESCRIPTION
**Pull request by davidlonjon**
Following quite a few experimentations with linting due to https://github.com/Automattic/jetpack/issues/20996 I realize that a few linting related configurations and standards had to be fixed with Boost.

#### Changes proposed in this Pull Request:

* Remove an old eslint configuration
* Add a new `.eslintignore` file to whitelist the `global.d.ts` file so it can be linted. Also adds it to the top of the Jetpack Monorepo  `.eslintignore`. This in effect disabling the rule to ignore every `d.ts` files from ( I think) the `@wordpress/eslint-plugin`. I am whitelisting it because it seems we missed a few issues from that file as it was not linted. I am not sure if there is a reason not to lint it.
* Fix the `global.d.ts` file by fixing an import path and removing an import that was not used anymore.
* Fix a couple of paths in the tsconfig file
* Fix a type in `app/assets/src/js/utils/generate-critical-css.ts` that somehow was missed.

#### Jetpack product discussion

N/A

#### Does this pull request change what data or activity we track or use?

N/A

#### Testing instructions:
Make sure the lining on the local environment still passes as well as the one on the CI.

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -40,6 +40,7 @@ projects/plugins/boost/app/assets/dist/**
 projects/plugins/boost/tests/**
 projects/plugins/boost/docker/**
 projects/plugins/boost/wordpress/**
+!projects/plugins/boost/app/assets/src/js/*.d.ts
```
Comment (anomiex) [inline .eslintignore:43]: I see there are two .d.ts files in the repo, `projects/plugins/boost/app/assets/src/js/global.d.ts` and `projects/plugins/boost/app/assets/src/js/svg/svgs.d.ts`. This rule only matches the first.

If you want to match both, you could do like
```suggestion
!projects/plugins/boost/app/assets/src/js/**/*.d.ts
```

And similarly in your local .eslintignore.

Comment (thingalon) [reply .eslintignore]: Yep, tested and the more globby path also catches svgs.d.ts. Nice catch! Committed in 91db7f771

**Review by thingalon — APPROVED** [review]
👍 works well, looks good to me. Thanks @davidlonjon !

File: .eslintignore (hunk 1 above)
Comment (davidlonjon) [reply .eslintignore]: Thank you @anomiex for taking a look at the PR and spotting this one 🙏🏻
Also thank you @thingalon 👍 

I have added the globby path to the local .eslintignore file too in commit 581e22441cb4c15b98c3b4fff78a7e2ece6cdcf0

**Review by thingalon — APPROVED** [review]
Still looking good!

**Review by anomiex — APPROVED** [review]
Seems ok to me, but needs a master merge.

You might consider whether you need local linting at all rather than making sure the monorepo's linting checks everything, since the local linting is not run in CI. But that's not something that needs to hold up this PR.

**Comment by davidlonjon** [comment]
> Seems ok to me, but needs a master merge.
> 
> You might consider whether you need local linting at all rather than making sure the monorepo's linting checks everything, since the local linting is not run in CI. But that's not something that needs to hold up this PR.

Thanks @anomiex. Yes, the next step would be to have all the lining done by the Monorepo's linting. I have just created now this https://github.com/Automattic/jetpack/issues/21066 for it. I will implement it after this other linting related [PR](https://github.com/Automattic/jetpack/pull/21026) fixing local linting is also merged.

**Pull request merged**